### PR TITLE
perf: Setting metadatabase

### LINKED ISSUE #1940: metadatabase is not set in the landing page.
### Issue Summary

The metadataBase property in metadata export is not set for resolving social open graph or twitter images, this doesn't affect the landing page but would be useful to set at the layout level

### Steps to Reproduce

1. Run pnpm build see the warning.

### Expected behavior

See https://nextjs.org/docs/app/api-reference/functions/generate-metadata#metadatabase

the warning should be removed by setting the dynamic URLs and then fallback

### PR DESCRIPTION
**Pull request by harshsbhat**
## What does this PR do?

It sets up the Metabase in the layout file for the landing page 

Fixes #1940

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

There won't be any Metadatabase warnings after running pnpm build

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

### CODE REVIEW

File: apps/www/app/layout.tsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ const PostHogPageView = dynamic(() => import("@/providers/posthog/PostHogPageVie
   ssr: false,
 });
 export const metadata: Metadata = {
+  metadataBase: new URL("https://unkey.com/"),
```
Comment (perkinsjr) [inline apps/www/app/layout.tsx:14]: When overriding the default, we recommend using environment variables to compute the URL. This allows configuring a URL for local development, staging, and production environments.

Setting this to our URL doesn't truly fix the issue, we should be reading from process.env.X and then setting it that way. 

Once this is set the images should be set to a relative path versus absolute.

**Review by perkinsjr — CHANGES_REQUESTED** [review]
Please lint code 

pnpm fmt 

please see comment.

File: apps/www/app/layout.tsx (hunk 1 above)
Comment (chronark) [reply apps/www/app/layout.tsx:14]: can we use our zod env to parse this and set a default?

```ts
baseUrl: z.string().default("https://unkey.com")
```

?

**Comment by harshsbhat** [comment]
> Please lint code 
> 
> pnpm fmt 
> 
> please see comment.

Yeah. I will work on the requested changes. I did run pnpm fmt. There were some problems in tools/migrate/main that are causing the linting issues.

**Comment by harshsbhat** [comment]
also can you check this #1938 PR too. Its a pretty quick fix.

File: apps/www/app/layout.tsx (hunk 1 above)
Comment (harshsbhat) [reply apps/www/app/layout.tsx:14]: image
this is how we are talking about it right? Also I think I should configure it to unkey URL if there is nothing in the .env

Comment (harshsbhat) [reply apps/www/app/layout.tsx:14]: > can

Like this right?

image

**Pull request closed without merging**